CONTRIBUTING.md

```diff
@@ -1,15 +1,158 @@
+             semi-colons, etc)
+- **refactor**: A code change that neither fixes a bug nor adds a feature
+- **perf**: A code change that improves performance
+- **test**: Adding missing tests
+- **chore**: Changes to the build process or auxiliary tools and libraries such as documentation
+generation
+- **revert**: reverts a previous commit. The subject should be the header of the reverted commit. In the body it should say: `This reverts commit <hash>.`, where the hash is the SHA of the commit being reverted.
+
+#### Scope
+The scope could be anything specifying place of the commit change. For example `$location`,
+`$browser`, `$compile`, `$rootScope`, `ngHref`, `ngClick`, `ngView`, etc...
+
+#### Subject
+The subject contains succinct description of the change:
+
+* use the imperative, present tense: "change" not "changed" nor "changes"
+* do not capitalize first letter
+* no dot (.) at the end
+
+#### Body
+Just as in the **subject**, use the imperative, present tense: "change" not "changed" nor "changes".
+The body should include the motivation for the change and contrast this with previous behavior.
+
+#### Footer
+The footer should contain any information about **Breaking Changes** and is also the place to
+reference GitHub issues that this commit **Closes**.
+
+**Breaking Changes** should start with the word `BREAKING CHANGE:` with a space or two newlines. The rest of the commit message is then used for this.
+
+### Submit a Pull Request
+
+
+1. Make your changes in a new git branch:
+
+```shell
+git checkout -b my-pr-branch master
+```
+2. Create your patch, **including appropriate test cases**.
+3. Commit your changes using a descriptive commit message that follows our
+    [commit message format](#Commit_Message_Format).
+
+```shell
+git commit -a
+```
+Note: the optional commit `-a` command line option will automatically "add" and "rm" edited files.
+
+
+4. Push your branch to GitHub:
+
+```shell
+git push --set-upstream origin my-pr-branch
+```
+
+5. In GitHub, target the pull request to `zanata:master`.
+6. If we suggest changes then:
+   a. Make the required updates.
+   b. Commit your changes to your branch (e.g. `my-pr-branch`).
+    * Push the changes to your GitHub repository (this will update your Pull Request).
+
+7. If the PR gets too outdated we may ask you to rebase and force push to
+update the pull request:
+```shell
+git rebase master -i
+git push origin my-pr-branch -f
+```
+
+*WARNING. Squashing or reverting commits and forced push thereafter may remove GitHub comments
+on code that were previously made by you and others in your commits.*
+
+That is it! Thank you for your contribution!
+
+### After your pull request is merged
+
+After your pull request is merged, you can safely delete your branch and
+pull the changes from the **master** (or **release** branch if you were
+working with release) branch of main (upstream) repository.
```

Comment:
It should be enough just to say it is safe to delete their branch - it is not really needed (they can keep the branch if they want). We should just link to documentation for if they don't know about deleting branches in git.